docs/articles/documentation/reference/test-api/testcontroller/README.md

```diff
@@ -0,0 +1,76 @@
+---
+layout: docs
+title: TestController
+permalink: /documentation/reference/test-api/testcontroller/
+---
+# TestController
+
+A *test controller* object exposes the test API's methods. The test controller is passed to each function that can run server-side test code (like [test](../global/test.md), [beforeEach](../fixture/beforeeach.md) or [afterEach](../fixture/aftereach.md)).
+
+Use the test controller to call [test actions](interact-with-the-page.md), handle [browser dialogs](interact-with-the-page.md), use the [wait function](interact-with-the-page.md), or [execute assertions](assert.md).
+
+```js
+fixture `My fixture`
+    .page `http://www.example.com/`;
+
+test('My Test', async t => {
+    await t
+        .setNativeDialogHandler(() => true)
+        .click('#populate')
+        .click('#submit-button');
+
+    const location = await t.eval(() => window.location);
+
+    await t.expect(location.pathname).eql('/testcafe/example/thank-you.html');
+});
+```
+
+The test runner also uses the test controller to access the internal context required for the test API to operate. When you call [selectors](select-page-elements.md) and [client functions](obtain-client-side-info.md) from Node.js callbacks, pass the test controller explicitly because the API cannot retrieve it from the context.
```

Comment:
```suggestion
The test runner also uses the test controller to access the internal context required for the test API to operate. When you call [selectors](select-page-elements.md) and [client functions](obtain-client-side-info.md) from Node.js callbacks, pass the test controller explicitly, because the API cannot retrieve it from the context.
```